### PR TITLE
Make remove incompatible processes more pessimistic

### PR DESCRIPTION
**Pull request by johscheuer**
# Description

*Please include a summary of the change and which issue is addressed. If this change resolves an issue, please include the issue number in the description.*

## Type of change

*Please select one of the options below.*

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- Documentation
- Other

## Discussion

*Are there any design details that you would like to discuss further?*

## Testing

*Please describe the tests that you ran to verify your changes. Unit tests?
Manual testing?*

*Do we need to perform additional testing once this is merged, or perform in a larger testing environment?*

## Documentation

*Did you update relevant documentation within this repository?*

*If this change is adding new functionality, do we need to describe it in our user manual?*

*If this change is adding or removing subreconcilers, have we updated the core technical design doc to reflect that?*

*If this change is adding new safety checks or new potential failure modes, have we documented and how to debug potential issues?*

## Follow-up

*Are there any follow-up issues that we should pursue in the future?*

*Does this introduce new defaults that we should re-evaluate in the future?*

### CODE REVIEW

File: controllers/remove_incompatible_processes.go [hunk 1]
```diff
@@ -99,9 +122,19 @@ func processIncompatibleProcesses(ctx context.Context, r *FoundationDBClusterRec
...
 
+		if address.Port == 0 {
+			logger.V(1).Info("Ignore incompatible connection with port 0", "address", address)
+			continue
+		}
+
 		incompatibleConnections[address.IPAddress.String()] = fdbv1beta2.None{}
 	}
 
+	coordinatorSet := make(map[string]fdbv1beta2.None)
```
Comment (johscheuer) [inline controllers/remove_incompatible_processes.go:133]: TODO: Add test case for his one.

File: controllers/remove_incompatible_processes.go [hunk 2]
```diff
@@ -90,6 +90,29 @@ func processIncompatibleProcesses(ctx context.Context, r *FoundationDBClusterRec
...
+		return err
+	}
+
+	if minimumUptime < float64(cluster.GetMinimumUptimeSecondsForBounce()) {
+		logger.V(1).Info("Skipping reconciler and waiting until cluster is for up minimum uptime")
+		return nil
+	}
+
+	// Ensure the cluster is running at fault tolerance before recreating Pods.
+	hasDesiredFaultTolerance, err := internal.HasDesiredFaultToleranceFromStatus(logger, status, cluster)
```
Comment (johscheuer) [inline controllers/remove_incompatible_processes.go:105]: TODO: Add test case for his one.

File: controllers/remove_incompatible_processes.go [hunk 3]
```diff
@@ -90,6 +90,29 @@ func processIncompatibleProcesses(ctx context.Context, r *FoundationDBClusterRec
 		return nil
 	}
 
+	// Wait until the cluster is running for the minimum uptime before looking for incompatible processes.
+	minimumUptime, _, err := internal.GetMinimumUptimeAndAddressMap(cluster, status, r.EnableRecoveryState)
```
Comment (johscheuer) [inline controllers/remove_incompatible_processes.go:94]: TODO: Add test case for his one.

File: controllers/remove_incompatible_processes_test.go [hunk 4]
```diff
@@ -32,7 +32,7 @@ import (
 	. "github.com/onsi/gomega"
 )
 
-var _ = Describe("restart_incompatible_pods", func() {
+var _ = FDescribe("restart_incompatible_pods", func() {
```
Comment (johscheuer) [inline controllers/remove_incompatible_processes_test.go:35]: Remove focus.

**Review by brownleej — APPROVED** [review]
👍

**Pull request merged**